### PR TITLE
MARP-616-A-Trust-Refactoring

### PR DESCRIPTION
**Pull request by nqhoan-axonivy**

### CODE REVIEW

File: a-trust-connector/src/resources/config/ATrust-OpenAPI.yaml
Comment (ivy-rew) [inline a-trust-connector/src/resources/config/ATrust-OpenAPI.yaml:1]: this is absolutely great, that you craft the openAPI for a-trust, this increases the maintainability of this service a lot :heart:

**Review by ivy-rew — COMMENTED** [review]
You completely had to switch the stack form SOAP to REST.
Now I do not see any tests that use mocks of the service payloads going to an fro.
Just like in the previous version it tests against live instances. We should avoid this and use a mock like in all other REST connectors. This will help us a lot:
- presales can show a demo when the service is not reachable or they are being offline
- we have documentation how the service worked in the past; if the service changes.

You may copy the 'mocking' approach from another connector or our examples: https://github.com/axonivy/project-build-examples/tree/master/compile-test/crmTests/src/ch/ivyteam/crm/mocks

File: a-trust-connector-product/README_DE.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+[A-Trust](https://www.a-trust.at) ist ein österreichisches Unternehmen, das elektronische Services, wie digitale Signaturen und Zertifikate, anbietet,
+um eine sichere Online-Kommunikation zu gewährleisten. Die **Signaturbox** von A-Trust ist eine Lösung für digitale Signaturen, mit der 
+Benutzer Dokumente sicher digital signieren können, um Authentizität (**Wer** signiert) und Integrität (**Was** wird signiert) zu gewährleisten.
+
+- Basiert auf REST Web Service Technologien.
+- Signiert PDFs.
+- Unterstützt dich mit einer  Demo-Implementierung, um deinen Integrationsaufwand zu reduzieren.
+- Ermöglicht es low code citizen developers bestehende Geschäftsprozesse mit Handy-Signaturfunktionen zu erweitern.
+
+Für die Nutzung von A-Trust benötigst Du einen Account, den Dir https://www.a-trust.at/de/produkte/qualifizierte_signaturservices/xidentity/ erstellt.
```
Comment (phhung-axonivy) [inline a-trust-connector-product/README_DE.md:10]: It does not match the English version.

File: a-trust-connector/processes/ATrust/StartSignature.p.json [hunk 2]
```diff
@@ -6,216 +6,188 @@
...
+            "BusinessDataInfo info = ivy.repo.save(in.signatureDocumentData);",
+            "ivy.task.customFields().stringField(ATrustCustomField.SIGNATURE_DOCUMENT_DATA_ID.getKey()).set(info.getId());",
+            "ivy.task.customFields().stringField(ATrustCustomField.SIGN_TICKET_ID.getKey()).set(in.signatureTicket);",
+            "ivy.task.customFields().stringField(RequestParam.CALLBACK_HASH.getKey()).set(in.callbackSafetyHash);",
+            "ivy.task.customFields().stringField(ATrustCustomField.SIGNATURE_JOB_ID.getKey()).set(ivy.repo.save(in.signatureJob).getId());",
             "",
-            "if (StringUtils.isEmpty(container.callbackUrlAfterSignature)) {",
-            "  container.callbackUrlAfterSignature = ivy.html.startRef(\"ATrust/EndSignature/SignatureCallBack.ivp\");",
+            "if (in.signatureJob.embeddedInframe) {",
+            "  in.forwardTargetUrl = in.forwardTargetUrl + ATrustService.generateIFrameParameters(450, 450, \"\", 0);",
```
Comment (phhung-axonivy) [inline a-trust-connector/processes/ATrust/StartSignature.p.json:85]: Why do you use the fixed value 450?

File: a-trust-connector/src/com/axonivy/connector/atrust/constant/Constants.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.axonivy.connector.atrust.constant;
+
+public class Constants {
+
+	public static final String SIGNED_PREFIX = "SIGNED_";
+	public static final String SEMICOLON = ";";
+	public static final String PDF = ".pdf";
+	public static final String TEMPLATE_DISPLAYNAME_PATTERN = "%s (#%d)";
```
Comment (phhung-axonivy) [inline a-trust-connector/src/com/axonivy/connector/atrust/constant/Constants.java:8]: `TEMPLATE_DISPLAYNAME_PATTERN` -> `TEMPLATE_DISPLAY_NAME_PATTERN `

**Review by phhung-axonivy — CHANGES_REQUESTED** [review]

**Comment by nqhoan-axonivy** [comment]
Thank @ivy-rew so much for your hints!
I'm working on it now. Just make everything back to work, now I will create a mock service for testing.

**Review by phhung-axonivy — APPROVED** [review]
LGTM

**Pull request merged**